doc.md

```diff
@@ -0,0 +1,34 @@
+`PartialOrd` Documentation
+==========================
+
+[Strict partial ordering relation](https://en.wikipedia.org/wiki/Partially_ordered_set#Strict_and_non-strict_partial_orders).
+
+This trait extends the partial equivalence relation provided by `PartialEq` (`==`) with `partial_cmp(a, b) -> Option<Ordering>`, which is a [trichotomy](https://en.wikipedia.org/wiki/Trichotomy_(mathematics)) of the ordering relation when its result is `Some`:
+
+* if `a < b` then `partial_cmp(a, b) == Some(Less)`
+* if `a > b` then `partial_cmp(a, b) == Some(Greater)`
+* if `a == b` then `partial_cmp(a, b) == Some(Equal)`
+
+and the absence of an ordering between `a` and `b` when `partial_cmp(a, b) == None`. Furthermore, this trait defines `<=` as `a < b || a == b` and `>=` as `a > b || a == b`.
+
+The comparisons must satisfy, for all `a`, `b`, and `c`:
+
+- _transitivity_: if `a < b` and `b < c` then `a < c`
+- _duality_: if `a < b` then `b > a`
+
+The `lt` (`<`), `le` (`<=`), `gt` (`>`), and `ge` (`>=`) methods are implemented in terms of `partial_cmp` according to these rules. The default implementations can be overridden for performance reasons, but manual implementations must satisfy the rules above.
+
+From these rules it follows that `PartialOrd` must be implemented symmetrically and transitively: if `T: PartialOrd<U>` and `U: PartialOrd<V>` then `U: PartialOrd<T>` and `T: PartialOrd<V>`.
```

Comment:
Not technically false, but from "`T: PartialOrd<U>` and `U: PartialOrd<V>`" follow not only "`U: PartialOrd<T>` and `T: PartialOrd<V>`", but also `V: PartialOrd<U>` and `V: PartialOrd<T>`. I think?

Reply:
@RalfJung noticed on IRC that this condition is not enough.

If `T: PartialOrd<U>`, let `a0: T`, `a1: T`, `b: U`. Then assume `a1 < b`  and `b < a2`. From transitivity it follows that `a1 < a2`, but we do not require `T: PartialOrd<T>` so this is currently not well-typed. 

That is, if `T != U`, and the user implements `T: PartialOrd<U>`, the user needs to implement `T: PartialOrd<T>`, `U: PartialOrd<T>`, and `U: PartialOrd<U>` accordingly for all requirements (like transitivity) to be fulfilled.

Iff the user then adds an implementation of `T: PartialOrd<V>`, then it would also need to implement `V: PartialOrd<V>`, `V: PartialOrd<T>`,  `U: PartialOrd<V>`, and `V: PartialOrd<U>`.